cdap-watchdog/src/main/java/io/cdap/cdap/logging/clean/LogCleaner.java

```diff
@@ -99,21 +94,16 @@ private void cleanupLogFolders() {
     LOG.info("Starting log folder cleanup");
     try {
       long startTime = System.currentTimeMillis();
-      LOG.debug("Starting log folder cleanup");
+      LOG.debug("Starting log folder cleanup {}", startTime);
       folderCleanupCount = 0;
 
-      // TODO: (CDAP-19437): Exclude recent log folders during cleanups in LogCleaner service.
-      // For now exclude folders for current date and current date - 1.
-      long currentTime = System.currentTimeMillis();
-      Set<String> excludeFolders = new HashSet<String>();
-      for (int day = 0; day < EXCLUDE_FOLDER_CLEANUP_DAYS; day++) {
-        long time = currentTime - day * MILLISECONDS_IN_DAY;
-        String date = LogFileManager.formatLogDirectoryName(time);
-        excludeFolders.add(date);
+      // Exclude folders which were last modified within retention period.
+      // Clean up stale empty folders inside logs folder.
+      long checkpointTimestamp = startTime - retentionDurationMs;
+      for (Location location : logsDirectoryLocation.list()) {
+        cleanupEmptyFolders(location, checkpointTimestamp);
```

Comment:
So the folder being deleted may not be empty, correct? If that's the case, please rename the method appropriately.

Reply:
Renamed to 'cleanupFoldersIfEmpty'